app/models/evaluation.rb

```diff
@@ -231,13 +246,18 @@ def self.to_csv(json = nil)
     json_params = json.nil? ? nil : JSON.parse(json)
     filter_params = json_params["_json"].nil? ? nil : json_params["_json"]
 
-    if filter_params.nil?
-      # this will never be reached currently, need to fix.
-      evaluations = Evaluation.order(id: :asc)
-    else
-      where_params = parse_filters(filter_params)
-      evaluations = run_csv_query(where_params)
+    where_statement = []
+    where_params = parse_filters(filter_params)
+    where_params.map do |k, v|
+      where_statement << where_fields(k,v) unless v.nil?
     end
-    generate_csv(evaluations)
+
+    where_statement = where_statement.empty? ? nil : where_statement.join(' AND ')
```

Comment:
@Levia I thought I'd force it to `nil` if the statement is empty.

Reply:
But I think you are right I need to change the other code instead to detect if it is empty!